File: sdks/ruby/rest/lib/Brume/types/readyz_response.rb
# frozen_string_literal: true

module Brume
  module Types
    # `GET /readyz` success body. The 200 / 503 distinction is what load
    # balancers key on; the body is for human debugging.
    class ReadyzResponse < Internal::Types::Model
      field :status, -> { String }, optional: false, nullable: false
    end
  end
end

File: sdks/php/rest/src/Types/ReadyzResponse.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;

/**
 * `GET /readyz` success body. The 200 / 503 distinction is what load
 * balancers key on; the body is for human debugging.
 */
class ReadyzResponse extends JsonSerializableType
{
    /**
     * @var string $status Always `"ready"`.
     */
    #[JsonProperty('status')]
    public string $status;

    /**
     * @param array{
     *   status: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->status = $values['status'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
